connection_notifier.cc

```diff
@@ -0,0 +1,59 @@
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "connection_notifier.hh"
+#include "db/query_context.hh"
+#include "cql3/constants.hh"
+
+#include <stdexcept>
+
+namespace db::system_keyspace {
+const char *const clients_cf_name();
+}
+
+sstring to_string(client_type clit) {
+    switch (clit) {
+        case client_type::cql: return "cql";
+        case client_type::thrift: return "thrift";
+        case client_type::alternator: return "alternator";
+        default: throw std::runtime_error("Invalid client_type");
+    }
+}
+
+future<> notify_new_client(client_data cd) {
+    const thread_local sstring req
+            = format("INSERT INTO system.{} (address, client_type, port, shard_id, protocol_version, username) "
+                     "VALUES (?, ?, ?, ?, ?, ?);", db::system_keyspace::clients_cf_name());
+    
+    return db::execute_cql(req,
+            std::move(cd.ip), to_string(cd.clit), cd.port, cd.shard_id,
+            cd.protocol_version.has_value() ? data_value(*cd.protocol_version) : data_value::make_null(int32_type),
```

Comment:
I'm unhappy about introducing tombstone here (that's what will happen, right?). Any ideas are welcome.

Reply:
There is no way around it. However, you can set gc_grace_second = 0 in the schema to reduce the time the tombstone hangs around (this is safe since this is a local table and will never be repaired)